eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -155,18 +154,10 @@ class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, wallet: OnChainA
         d.channels.get(TemporaryChannelId(msg.temporaryChannelId)) match {
           case None =>
             val channelConfig = ChannelConfig.standard
-            val defaultChannelType = ChannelTypes.pickChannelType(d.localFeatures, d.remoteFeatures)
-            val chosenChannelType: Either[InvalidChannelType, SupportedChannelType] = msg.channelType_opt match {
-              case None => Right(defaultChannelType)
-              case Some(proposedChannelType: UnsupportedChannelType) => Left(InvalidChannelType(msg.temporaryChannelId, defaultChannelType, proposedChannelType))
-              case Some(proposedChannelType: SupportedChannelType) =>
-                // We ensure that we support the features necessary for this channel type.
-                val featuresSupported = proposedChannelType.features.forall(f => d.localFeatures.hasFeature(f))
-                if (featuresSupported) {
-                  Right(proposedChannelType)
-                } else {
-                  Left(InvalidChannelType(msg.temporaryChannelId, defaultChannelType, proposedChannelType))
-                }
+            val defaultChannelType = ChannelTypes.defaultFromFeatures(d.localFeatures, d.remoteFeatures)
+            val chosenChannelType: Either[InvalidChannelType, SupportedChannelType] = ChannelTypes.negotiateChannelType(d.localFeatures, defaultChannelType, msg.channelType_opt) match {
+              case Left(rejectedChannelType) => Left(InvalidChannelType(msg.temporaryChannelId, defaultChannelType, rejectedChannelType))
+              case Right(r) => Right(r)
```

Comment:
@t-bast I wonder why we don't handle the `channel_type` negotiation, as fundee, in `Helpers.validateParamsFundee`? That would be more consistent with the funder workflow, even if we would create a channel actor for nothing (that's what we do for all other parameters).

Reply:
We have to do it here, because we need to know the channel type *before* creating the channel (calling `createNewChannel`), since we need to know whether we must ask `bitcoind` for a static payment point (`static_remotekey`).

It could be refactored though to move the creation of the payment basepoint to the `Channel`, but I'm not sure it's trivial given how we instantiate `ChannelParams` (and if we do that, we should move the funder case as well).